pkg/rest/rest.go

```diff
@@ -25,14 +25,36 @@ type Endpoint struct {
 }
 
 func NewEndpoint(input string) (*Endpoint, error) {
-	uri, err := parseRestfulURI(input)
+	uri, err := url.ParseRequestURI(input)
 	if err != nil {
 		return nil, err
 	}
 	scheme, err := toRestScheme(uri.Scheme)
 	if err != nil {
 		return nil, err
 	}
+	switch scheme {
+	case TCP:
+		if len(uri.Host) < 1 {
```

Comment:
`toRestScheme` is only called once after this change, and it is mostly checking that the uri scheme is a valid one, and then in this `switch`/`case`, we do more validation on the URI.
I’d move `toRestScheme` and the validation code from `parseRestfulURI` to a single `validateURI(uri)` method which would do everything at once.
I prefer to have the validation code in a separate function to avoid cluttering `NewEndpoint`.

Reply:
I've removed the third commit.